src/design-system/components/Text/Text.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { ElementRef, forwardRef, ReactNode, useMemo } from 'react';
+import { Text as NativeText } from 'react-native';
+import {
+  CustomColor,
+  useForegroundColor,
+} from '../../color/useForegroundColor';
+import { createLineHeightFixNode } from '../../typography/createLineHeightFixNode';
+import {
+  nodeHasEmoji,
+  nodeIsString,
+  renderStringWithEmoji,
+} from '../../typography/renderStringWithEmoji';
+import { textSizes, textWeights } from '../../typography/typography';
+
+const validColors = [
+  'accent',
+  'action',
+  'primary',
+  'secondary',
+  'secondary30',
+  'secondary40',
+  'secondary50',
+  'secondary60',
+  'secondary70',
+  'secondary80',
+] as const;
+
+export type TextProps = {
+  align?: 'center' | 'left' | 'right';
+  color?: typeof validColors[number] | CustomColor;
+  numberOfLines?: number;
+  size?: keyof typeof textSizes;
+  tabularNumbers?: boolean;
+  testID?: string;
+  uppercase?: boolean;
+  weight?: keyof typeof textWeights;
+} & (
+  | {
+      containsEmoji: true;
+      children: string | (string | null)[];
+    }
+  | { containsEmoji?: false; children: ReactNode }
+);
+
+/**
+ * @description Default size is `"16px"`
+ */
+export const Text = forwardRef<ElementRef<typeof NativeText>, TextProps>(
+  function Text(
+    {
+      numberOfLines,
+      containsEmoji: containsEmojiProp = false,
+      children,
+      testID,
+      align: textAlign,
+      color = 'primary',
+      size = '16px',
+      weight = 'regular',
+      tabularNumbers = false,
+      uppercase = false,
+    },
+    ref
+  ) {
+    if (__DEV__) {
```

Comment:
I'm thinking of this warning. There are fields that are simply unknown (come from the API), e.g. descriptions of NFTs. Then I don't want to see the warning everywhere. Can we introduce "maybe" option?

Reply:
Yeah, I wasn't sure about this warning when I wrote it. I'll revisit this.

Reply:
I'm switching this to a console.log instead of throwing errors. I think it's important that these get picked up to ensure that the layout is stable when emoji are rendered, but in hindsight it's pretty heavy handed to blow up the entire render over it.

The naming maybe doesn't make it as clear as it could, but `containsEmoji` should be set to `true` in cases where dynamic text _might_ contain emoji. Tried to think of a better name and failed, if you anyone has any suggestions? 😅